packages/devtools_app/lib/src/screens/vm_developer/object_inspector_view_controller.dart

```diff
@@ -2,22 +2,68 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:collection/collection.dart';
 import 'package:flutter/foundation.dart';
 import 'package:vm_service/vm_service.dart';
 
 import '../../primitives/auto_dispose.dart';
+import '../../shared/globals.dart';
+import '../debugger/program_explorer_controller.dart';
 import 'object_viewport.dart';
 import 'vm_object_model.dart';
 
 /// Stores the state information for the object inspector view related to
 /// the object history and the object viewport.
 class ObjectInspectorViewController extends DisposableController
     with AutoDisposeControllerMixin {
+  ObjectInspectorViewController() {
+    // addAutoDisposeListener(serviceManager.isolateManager.selectedIsolate,
+    //     () async {
+    //   restartController();
+    // });
+
+    addAutoDisposeListener(
+      scriptManager.sortedScripts,
+      selectAndPushMainScript,
+    );
+
+    objectHistory.current.addListener(_onCurrentObjectChanged);
+  }
+
+  final programExplorerController = ProgramExplorerController()
+    ..initialize()
+    ..initListeners();
+
   final objectHistory = ObjectHistory();
 
+  Isolate? isolate;
+
+  final _currentScriptRef = ValueNotifier<ScriptRef?>(null);
+
+  ValueListenable<ScriptRef?> get currentScriptRef => _currentScriptRef;
+
   ValueListenable<bool> get refreshing => _refreshing;
   final _refreshing = ValueNotifier<bool>(false);
 
+  Future<void> _onCurrentObjectChanged() async {
+    final currentObjectValue = objectHistory.current.value;
+
+    _currentScriptRef.value = currentObjectValue?.scriptRef;
+
+    if (currentObjectValue != null) {
+      await programExplorerController
+          .selectScriptNode(currentObjectValue.scriptRef);
+
+      if (objectHistory.current.value?.outlineNode != null) {
+        programExplorerController
+          ..selectOutlineNode(currentObjectValue.outlineNode!)
```

Comment:
Nit: raise `currentObjectValue.outlineNode!` into a local variable to avoid the extra null assertions.

Reply:
Thanks for the tip